### PR TITLE
Implementation of Corda Branch

### PR DESCRIPTION
**Pull request by alokrajiv**
Referencing discussion offline as well as over here on #324 

Can continue any recommendations here or on discussions.

### CODE REVIEW

**Comment by ami000** [comment]
@thakurat44 @Dimple-Kanwar marking it as ready for review here.

**Comment by rjchow** [comment]
Hey guys sorry, I've been in and out of office and missed this PR - will look at this in the coming sprint

File: src/common/contexts/provider.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import React, { createContext, useContext, useEffect, useState } from "react";
 import { INFURA_API_KEY, NETWORK_NAME } from "../../config";
 
 const getProvider =
-  NETWORK_NAME === "local"
+  NETWORK_NAME === "local" || NETWORK_NAME === "Corda Enterprise"
     ? new providers.JsonRpcProvider()
```
Comment (rjchow) [inline src/common/contexts/provider.tsx:8]: what is supposed to happen when it is set to "Corda Enterprise" ?

Comment (rjchow) [reply src/common/contexts/provider.tsx:8]: @alokrajiv

**Comment by ami000** [comment]
Hi @rjchow
The below points outline the changes when NETWORK_NAME is set to **Corda Enterprise**

- When network name is set to “Corda Enterprise”, the trade trust website gets connected to TRAMES Corda Enterprise network instead of Ethereum network.
- In the Corda Enterprise network, the TradeTrust website will be using only the connection provider(ConnectedRouter) instead of other providers like TokenInformationContextProvider, ProviderContextProvider.
- In the case of Corda Enterprise, a custom verifier is created for document verification which calls TRAMES verifier API.
- In the Corda Enterprise network, verification fragments are hashValid, issuedValid, consumedValid, identityValid, revokedValid and DNS verification as well. In the Ethereum network, verification fragments are hashValid, issuedValid , identityValid, revokedValid.

- Changes in document status
  - Document status have a check for Corda Enterprise which allow us to see more detailed state of a TT document
  - We have three statuses of the document which we are showing on the document viewer
    - Issued
    - Revoked
    - Consumed
- Changes in TT Document Viewer
  - We have added a custom document renderer to render TT documents from Cords Enterprise

**Comment by rjchow** [comment]
@ami000 , how exactly does it connect? The only thing I see in the PR is that it uses the local jsonrpc provider - how can I test that out?

File: src/common/contexts/provider.tsx (hunk 1 above)
Comment (alokrajiv) [reply src/common/contexts/provider.tsx:8]: hi @rjchow - just reminded @ami000 . He has given a detailed explanation on this. Let us know if any clarifications, we can open a discussion or we can continue here :) ?

Comment (rjchow) [reply src/common/contexts/provider.tsx:8]: Ah I replied in the main thread because @ami000 did over there

**Comment by ami000** [comment]
@rjchow, the only way to connect to the Corda network is through the API being called in the custom verifier.

jsonRpcProvider is being called only when the network name is either local or Corda. So if Network is local then Ethereum's local configuration will be used but in the case of Corda, jsonRPcProvider will be called but will do nothing, as the Ethereum's configuration is different from Corda.

This line is maintained so as to not break any local runs.

**Comment by rjchow** [comment]
@ami000 is `https://chewbacca-dev.blockchain.trames-engineering.com/document/validateTTDocument` the main Corda network that is publicly accessible and writable by anyone?

**Comment by ami000** [comment]
@rjchow 
Corda is a public but permissioned network. However, the above endpoint is sponsored and hosted for free by us for the community (anyone else can as well) through our node for public reading.
For writing or reading by themselves, you can host a node by approaching the Non-Profit Corda foundation that dispenses identities.
Also note, similar to Ropstein this network is our *test network* but in production will cut over to the public Corda network that you mentioned.

**Comment by rjchow** [comment]
@ami000 @alokrajiv I understand that at a higher level we have aligned our approach to maintain the Corda gateway as a standalone instance instead of merging it into tradetrust-website, so I will be closing this pull request

**Pull request closed without merging**